x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrRetentionLeases.java

```diff
@@ -17,8 +17,22 @@
      * @param leaderUUID   the leader index UUID
      * @return the retention lease ID
      */
-    public static String retentionLeaseId(final String followerUUID, final String leaderUUID) {
-        return String.format(Locale.ROOT, "%s-following-%s", followerUUID, leaderUUID);
+    public static String retentionLeaseId(
+            final String localClusterName,
+            final String remoteClusterName,
+            final String followerIndexName,
```

Comment:
perhaps pass an `Index` object here for the follower and the leader index

Reply:
I pushed a commit for this.